transaction/core/src/tx.rs

```diff
@@ -452,6 +452,14 @@ impl ReprBytes for TxOutConfirmationNumber {
     }
 }
 
+impl From<&Vec<u8>> for TxOutConfirmationNumber {
+    fn from(src: &Vec<u8>) -> Self {
```

Comment:
This would fail if the vector isn't 32 bytes. I think this From impl should be removed, and instead wherever its used should check the length, copy into a slice and use the `impl core::convert::AsRef<[u8; 32]> for TxOutConfirmationNumber {` implementation

Reply:
I removed this From and moved the copy to the calling function, along with a length check